crates/issue/src/lib.rs

```diff
@@ -301,13 +305,75 @@ impl<T: Config> Pallet<T> {
         T::TreasuryPalletId::get().into_account_truncating()
     }
 
+    // Function to cancel an issue request and slash collateral
+    pub fn cancel_issue_request_and_slash_collateral(issue_id: &H256) -> Result<BalanceOf<T>, DispatchError> {
```

Comment:
It can be covered  `_cancel_issue` but then `_cancel_issue`  will have two paths

```Rust
fn _cancel_issue(issue_id: H256) -> Result<(), DispatchError> {
      if issue_id.is_replace_issue() {
              // handle replace logic of griefing collateral transfer from old_vault account to new vault
      }
      else {
             // handle slashing and decrease to-be-issued tokens
      }
     // set issue status as cancelled 
     // Deposit event
}

```